api/src/privacy/domain/usecases/anonymize-user.usecase.js

```diff
@@ -0,0 +1,93 @@
+import { UserAnonymizedEventLoggingJob } from '../../../identity-access-management/domain/models/UserAnonymizedEventLoggingJob.js';
+import { config } from '../../../shared/config.js';
+import { withTransaction } from '../../../shared/domain/DomainTransaction.js';
+
+/**
+ * @param params
+ * @param{string} params.userId
+ * @param{string} params.anonymizedByUserId
+ * @param{string} params.client
+ * @param{UserRepository} params.userRepository
+ * @param{AuthenticationMethodRepository} params.authenticationMethodRepository
+ * @param{MembershipRepository} params.membershipRepository
+ * @param{CertificationCenterMembershipRepository} params.certificationCenterMembershipRepository
+ * @param{OrganizationLearnerRepository} params.organizationLearnerRepository
+ * @param{RefreshTokenRepository} params.refreshTokenRepository
+ * @param{ResetPasswordDemandRepository} params.resetPasswordDemandRepository
+ * @param{UserLoginRepository} params.userLoginRepository
+ * @param{AdminMemberRepository} params.adminMemberRepository
+ * @param{UserAnonymizedEventLoggingJobRepository} params.userAnonymizedEventLoggingJobRepository
+ * @returns {Promise<null>}
+ */
```

Comment:
Mise à jour de la JS doc
```suggestion
/**
 * @param params
 * @param{string} params.userId
 * @param{string} params.anonymizedByUserId
 * @param{string} params.anonymizedByUserRole
 * @param{string} params.client
 * @param{UserRepository} params.userRepository
 * @param{AuthenticationMethodRepository} params.authenticationMethodRepository
 * @param{MembershipRepository} params.membershipRepository
 * @param{CertificationCenterMembershipRepository} params.certificationCenterMembershipRepository
 * @param{OrganizationLearnerRepository} params.organizationLearnerRepository
 * @param{RefreshTokenRepository} params.refreshTokenRepository
 * @param{ResetPasswordDemandRepository} params.resetPasswordDemandRepository
 * @param{UserLoginRepository} params.userLoginRepository
 * @param{UserAnonymizedEventLoggingJobRepository} params.userAnonymizedEventLoggingJobRepository
 * @returns {Promise<void>}
 */
```